src/content/docs/guides/data-shaping/transform-basic-values.mdx

```diff
@@ -0,0 +1,441 @@
+set word = capitalize(word)
+```
+
+```tql
+{word: "Hello", phrase: "good morning"}
+{word: "World", phrase: "how are you"}
+```
+
+## Mathematical operations
+
+Perform calculations on numeric values within your events.
+
+### Basic arithmetic
+
+Use standard arithmetic operators:
+
+```tql
+from {a: 10, b: 3},
+     {a: 20, b: 4}
+set sum = a + b,
+    diff = a - b,
+    product = a * b,
+    quotient = int(a / b)
+```
+
+```tql
+{a: 10, b: 3, sum: 13, diff: 7, product: 30, quotient: 3}
+{a: 20, b: 4, sum: 24, diff: 16, product: 80, quotient: 5}
+```
+
+### Rounding numbers
+
+Round numbers to specific precision:
+
+```tql
+from {value: 3.14159},
+     {value: 2.71828}
+set rounded = round(value),
+    ceil_val = ceil(value),
+    floor_val = floor(value)
+```
+
+```tql
+{value: 3.14159, rounded: 3, ceil_val: 4, floor_val: 3}
+{value: 2.71828, rounded: 3, ceil_val: 3, floor_val: 2}
+```
+
+### Mathematical functions
+
+Use [`abs()`](/reference/functions/abs) for absolute values and
+[`sqrt()`](/reference/functions/sqrt) for square roots:
+
+```tql
+from {x: -5, y: 16},
+     {x: -10, y: 25}
+set abs_x = abs(x), sqrt_y = sqrt(y)
+```
+
+```tql
+{x: -5, y: 16, abs_x: 5, sqrt_y: 4.0}
+{x: -10, y: 25, abs_x: 10, sqrt_y: 5.0}
+```
+
+## Working with null values
+
+Handle missing or null values gracefully in your data.
+
+### Provide default values
+
+Use the `else` keyword to replace null values:
+
+```tql
+from {name: "alice", age: 30},
+     {name: "bob"},
+     {name: "charlie", age: 25}
+set age = age? else 0,
+    status = status? else "unknown"
+```
+
+```tql
+{name: "alice", age: 30, status: "unknown"}
+{name: "bob", age: 0, status: "unknown"}
+{name: "charlie", age: 25, status: "unknown"}
+```
+
+## Generate values
+
+Create new values using generator functions:
+
+### Generate unique identifiers
+
+Use [`uuid()`](/reference/functions/uuid) to create unique identifiers:
+
+```tql
+from {user: "alice", action: "login"},
+     {user: "bob", action: "create"}
+set event_id = uuid(),
+    session_id = uuid()
+```
+
+```tql
+{user: "alice", action: "login", event_id: "a1b2c3d4-e5f6-7890-abcd-ef1234567890", session_id: "f9e8d7c6-b5a4-3210-fedc-ba0987654321"}
+{user: "bob", action: "create", event_id: "12345678-90ab-cdef-1234-567890abcdef", session_id: "abcdef12-3456-7890-abcd-ef1234567890"}
+```
+
+### Generate random numbers
+
+Use [`random()`](/reference/functions/random) to generate random values:
+
+```tql
+from {}, {}
+set random_float = random(),
+    random_int = int(random() * 100),
+    random_choice = if random() < 0.5 then "heads" else "tails"
+```
+
+```tql
+{random_float: 0.7394, random_int: 42, random_choice: "tails"}
+{random_float: 0.2156, random_int: 21, random_choice: "heads"}
+```
+
+### Access environment variables
+
+Use [`env()`](/reference/functions/env) to read environment variables:
+
+```tql
+from {}
+set home_dir = env("HOME"),
+    shell = env("SHELL"),
+    custom_var = env("MY_APP_CONFIG")? else "/default/config"
```

Comment:
The `?` after a function call is invalid syntax. Also, is this really "generating values"?

Reply:
I've improved that section.